### ISSUE TITLE
extension-link became incompatible with some webviews

### ISSUE THREAD

**Issue by martinstoeckli** [ueberdosis/tiptap#2876]
### What’s the bug you are facing?

I get a syntax error when a tiptap editor instance is created, if the extension-link is part of the build. The error only occurs when the editor is shown in a webview with given versions.

I hunted down the problem and could find the reason, there was a change between @tiptap/extension-link 2.0.0-beta.36 and 2.0.0-beta.37:

```
// 2.0.0-beta.36
parseHTML() {
    return [
        { tag: 'a[href]' },
    ];
},

// 2.0.0-beta.37
parseHTML() {
    return [
        { tag: 'a[href]:not([href *= "javascript:" i])' },
    ];
},
```

This syntax seems to be accepted only on newer browsers, see my tests below.

### Which browser was this experienced in? Are any special extensions installed?

Made some tests with different versions:

UWP WebView (most current Windows 10 version) ➽ failed
Android 6 WebView ➽ failed
Android 7 WebView ➽ failed
Android 11 WebView ➽ worked
Android 12 WebView ➽ worked

It seems that the syntax is accepted in newer browsers, though afaik UWP applications are stuck with their webview.

### How can we reproduce the bug on our side?

Just create a TipTap editor using the "link-extension" (most current version 2.0.0-beta.38) and show it in one of the mentioned WebViews. The editor won't be created, instead the creation is aborted with the error message "syntax error". The instantiation itself fails, it is not necessary to load content with a link.

If the link-extension is not part of the build, or if the tag is replaced with the old one `tag: 'a[href]'`, then the editor is working correctly, this is reproducable.

### What did you expect to happen?

The editor should be instantiated and shown, instead the creation fails with an exception.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by bdbch** [comment]
Yeah this change was made to avoid people implementing javascript links to minimize potential security issues.

I guess we have to do a compatibility check in parseHTML and if those :not selectors etc. are not available handle this parsing a bit different. Thanks for letting us know!

**Comment by martinstoeckli** [comment]
You are welcome, your work is really amazing. I think the problem cannot be overcome with a compatibility check at runtime, if this syntax exists, the parsing of the whole bundled JavaScript fails, it is not just when parseHtml is called.

**Comment by martinstoeckli** [comment]
Should I keep this open, and if so, how (cannot edit labels)?

**Comment by bdbch** [comment]
The bot usually reopens the ticket when you comment on it! I added it to our todo

**Comment by edenhermelin** [comment]
Any update on this one? 
this issue still happens, looks like the selector doesn't work 😞 
```
instrument.js:107 Unknown CSS selector action any
(anonymous) @ instrument.js:107
handleRules @ vcss.ts:59
(anonymous) @ vcss.ts:71
handleRules @ vcss.ts:70
matchSelector @ vcss.ts:89
matches @ vdom.ts:323
matches @ index.js:3135
matchTag @ index.js:2534
addElement @ index.js:2795
addDOM @ index.js:2748
addAll @ index.js:2909
addElementByRule @ index.js:2895
addElement @ index.js:2821
addDOM @ index.js:2748
addAll @ index.js:2909
addElementByRule @ index.js:2895
addElement @ index.js:2821
addDOM @ index.js:2748
addAll @ index.js:2909
addElementByRule @ index.js:2895
addElement @ index.js:2821
addDOM @ index.js:2748
addAll @ index.js:2909
addElementByRule @ index.js:2895
addElement @ index.js:2821
addDOM @ index.js:2748
addAll @ index.js:2909
addElementByRule @ index.js:2895
addElement @ index.js:2821
addDOM @ index.js:2748
addAll @ index.js:2909
parse @ index.js:2512
```

**Comment by bdbch** [comment]
As long as we don't have a good solution integrated into tiptap itself I'd suggest whenever you need this for an older webview try to extend the link extension and set the parseHTML function yourself:

```
Link.extend({
  parseHTML() {
    return [
      { tag: 'a[href]' }
    ]
  }
})
```

but as a warning - this will allow users to add links with integrated javascript again.

**Comment by cedricraud** [comment]
I'm experiencing the same warning in Node.js `Unknown CSS selector action any`, caused by the `[href *= "javascript:"` of the Link extension.

On non-browser environments, Prosemirror relies on `zeed-dom` to manipulate the DOM. This [line](https://github.com/holtwick/zeed-dom/blob/master/src/vcss.ts#L73) is generating a warning because it doesn't know how to handle the `*=` part of the selector. If we look deeper, we see that zeed-dom relies on the `css-what` library to parse CSS selectors and [their documentation](https://github.com/fb55/css-what#api) does not mention `*=` in the list of supported selectors. 🤷‍♂️

Maybe you should switch to the `^=` (Start) selector instead of `*=` (Contain) which is [supported](https://github.com/fb55/css-what/blob/master/readme.md?plain=1#L45) by the library?

**Comment by actraiser** [comment]
I encountered this warning when migrating millions of posts from an old discussion board to Tiptap using the generateJSON() function of @tiptap/html. If you use any function that triggers the warning on the server side, you can work around this annoyance by disabling console.error. This is of course not a fix by all means, but prevents messing up my terminal when processing batches.

```javascript
// Save the original console.warn function
const originalWarn = console.warn;

// Redirect console.warn to an empty function
console.warn = function () {};
             
// this generates the "Unknown CSS selector" warning, e.g.
const json = generateJSON($.html(), [
            Document,
            Paragraph,
            Text,
            Link,
            Strike,
            Italic,
            Bold,
            Blockquote,
          ]);

// restore the original function
console.warn = originalWarn;
``` 

Greets
-act

**Comment by bdbch** [comment]
This should've been fixed already in 738c436a - feel free to comment again if you think this is still a problem.
